enhancements/testing/ephemeral-agent-installer-e2e-tests.md

```diff
@@ -0,0 +1,373 @@
+
+#### Build
+
+The scripts/automation for building the agent-based installer should be inside the repository
+where the installer is being worked on and not dev-scripts. We will reference the built 
+container images and ISO in dev-scripts from those repositories. Currently this project is being 
+worked on in the  openshift-agent-team/fleeting repository but this may change in the future.
+
+#### Deployment
+
+Deploying the agent-based installer will be heavily linked to the environment set up. This
+logic will land in dev-scripts alongside the pre environment setup steps.
+
+#### Tests
+
+For the actual test scenarios themselves we should use Ginkgo/Gomega. This is 
+similar to the rest of Openshift. Well most of (if not all) the product is written in Golang
+so it makes sense to use a Golang framework for these.
+
+There is a repository created as a preliminary place to put these. Plan to use a similar
+structure to openshift-tests in this repository.
+
+We will be including existing e2e tests that have been created to validate the OCP cluster
+installed by the new agent based installer.
+
+#### Teardown
+
+Currently there is a `make clean` flow for starting new locally. We will need to add similar
+functionally for the flows that will be used for the agent based installer
+
+### Proposed Refactored Structure
+
+The following is describing general pattern of how to organize these things in the dev-scripts 
+repository. This will be added alongside the existing code.
+
+**Folder Structure:**
+
+```markdown
+playbooks/
+  example-e2e-workflow.yml
+roles/
+    common/
+        tasks/
+        handlers/
+        library/
+        files/
+        templates/
+        vars/
+        defaults/
+        meta/
+    example_role/
+        tasks/
+        defaults/
+        meta/
+        templates/
+        vars/
+```
+
+This is pretty open ended and meant to be. I just believe that focusing on making reusable roles 
+here for the steps is key. With this structure the aim is to be able to easily moving our roles into
+[Ansible Galaxy Modules](https://docs.ansible.com/ansible/latest/galaxy/dev_guide.html) if there is 
+desire for other projects and teams to use them in the future.
+
+**Playbook Structure:**
+
+Use the `pre`, `test`, and `post` pattern in Openshift CI locally in the playbooks and include 
+the playbooks. This will create a structure that will mirror the Openshift CI Step Registry
+and will provide clarity on what code is being run in each step.
+
+```markdown
+# File structure:
+playbooks/
+  example-e2e-workflow.yml
+  example-e2e-workflow_pre.yml
+  example-e2e-workflow_test.yml
+  example-e2e-workflow_post.yml
+
+# example-e2e-workflow.yml
+- hosts: localhost
+
+- name: Run e2e-workflow Pre steps
+  import_playbook: example-e2e-workflow_pre.yml
+
+- name: Run e2e-workflow Test steps
+  import_playbook: example-e2e-workflow_test.yml
+
+- name: Run e2e-workflow Post steps
+  import_playbook: example-e2e-workflow_test.yml
+```
```

Comment:
I'm missing the reason for this playbook (which seems to overlap somehow the ci-operator)? Why do you need this full stuff locally and not reuse all the resources provided by the CI?
Note that in any case it's possible to simply run the e2e locally via `openshift-tests`, once a cluster has been deployed (and I guess it has alreay been assumed that dev-scripts will cover the cluster deployment phase somehow)